src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1048,6 +1048,32 @@
   (jdbc/do-commands
     "CREATE INDEX idx_certnames_latest_report_id on certnames(latest_report_id)"))
 
+(defn add-producer-to-reports-catalogs-and-factsets
+  []
+  (jdbc/do-commands
+    "CREATE TABLE producers (
+        id bigint PRIMARY KEY,
+        name text NOT NULL UNIQUE)"
+    "CREATE SEQUENCE producers_id_seq
+        START WITH 1
+        INCREMENT BY 1
+        NO MINVALUE
+        NO MAXVALUE
+        CACHE 1"
```

Comment:
there's shorthand for some of the stuff in here.

* the sequence can be created with "create sequence producers_id_seq cycle"
* the ONLY clauses in the alter tables aren't needed
* the ::regclass in the call to nextval isn't needed
* the table can be created with the jdbc function "create-table-ddl"

Nothing critical, just note that the first migration in this file was actually generated by pg_dump or something rather than written, and is a bit different from how we write these generally.